environment/src/main/java/com/sequenceiq/environment/environment/flow/creation/handler/ResourceGroupCreationHandler.java

```diff
@@ -0,0 +1,168 @@
+        this.azureSingleResourceGroupFeatureSwitch = azureSingleResourceGroupFeatureSwitch;
+        this.costTagging = costTagging;
+        this.eventBus = eventBus;
+        this.environmentService = environmentService;
+        this.parameterService = parameterService;
+        this.clock = clock;
+    }
+
+    @Override
+    public void accept(Event<EnvironmentDto> environmentDtoEvent) {
+        EnvironmentDto environmentDto = environmentDtoEvent.getData();
+        environmentService.findEnvironmentById(environmentDto.getId())
+                .ifPresentOrElse(environment -> {
+                            try {
+                                createResourceGroup(environmentDto, environment);
+                                goToNetworkCreationState(environmentDtoEvent);
+                            } catch (Exception e) {
+                                goToFailedState(environmentDtoEvent, e.getMessage());
+                            }
+                        }, () -> goToFailedState(environmentDtoEvent, String.format("Environment was not found with id '%s'.", environmentDto.getId()))
+                );
+    }
+
+    private void createResourceGroup(EnvironmentDto environmentDto, Environment environment) {
+        if (!azureSingleResourceGroupFeatureSwitch.isActive()) {
+            LOGGER.debug("Azure single resource group feature turned off, not creating resourcegroup.");
+            return;
+        }
+        Optional<AzureResourceGroupDto> azureResourceGroupDtoOptional = getAzureResourceGroupDto(environmentDto);
+        if (!AZURE.name().equals(environmentDto.getCloudPlatform()) || azureResourceGroupDtoOptional.isEmpty()) {
+            LOGGER.debug("Cloudplatform not azure or no azure resource group dto defined, not creating resource group.");
+            return;
+        }
+        AzureResourceGroupDto azureResourceGroupDto = azureResourceGroupDtoOptional.get();
+        if (!CREATE_NEW.equals(azureResourceGroupDto.getResourceGroupCreation())) {
+            LOGGER.debug("Azure resource group creation not requested, not creating resource group.");
+            return;
+        }
+        CloudCredential cloudCredential = credentialToCloudCredentialConverter.convert(environmentDto.getCredential());
+        AzureClient azureClient = azureClientService.getClient(cloudCredential);
+
+        String resourceGroupName = String.format("%s_%d", environmentDto.getName(), clock.getCurrentTimeMillis());
+        LOGGER.debug("Azure generated resource group name: {}.", resourceGroupName);
+        azureClient.createResourceGroup(resourceGroupName, environmentDto.getLocation().getName(), mergeTags(environmentDto));
+        parameterService.updateResourceGroupName(environment, resourceGroupName);
+        LOGGER.debug("Azure resource group created successfully.");
+    }
+
+    private Map<String, String> mergeTags(EnvironmentDto environmentDto) {
+        CDPTagMergeRequest mergeRequest = CDPTagMergeRequest.Builder
```

Comment:
maybe ewe should just put this into the DTO directly, in that case we can just use that object

Reply:
Yes, I was wondering myself why is the combination of different tags done outside of the dto.